src/test-utils/smoke-test-utils.ts

```diff
@@ -40,25 +71,38 @@ export const isDataPlaneRequest = (response, targetUrl) => {
 
 /** Returns true when all events were ingested */
 export const verifyIngestionWithRetry = async (
-    rumClient,
-    eventIds,
-    timestamp,
-    monitorName,
-    retryCount
+    rumClient: RUMClient,
+    eventIds: any[],
+    timestamp: number,
+    monitorName: string | undefined,
+    retryCount: number,
+    metadataAttributes: string[] | undefined = undefined
 ) => {
     while (true) {
         if (retryCount === 0) {
             console.log('Retry attempt exhausted.');
             return false;
         }
         try {
-            await isEachEventIngested(
+            const actual: Map<string, string[]> = await getIngestedEvents(
                 rumClient,
-                eventIds,
                 timestamp,
                 monitorName
             );
-            return true;
+
+            return (
```

Comment:
nit: Could we refactor this so that the boolean is returned by a single method, and the single method contains the three different methods checking for 1) whether event is ingested, 2) whether valid attributes are there, 3) whether invalid attributes are there?

Reply:
Later on, if we introduce another condition to check, the original `verifyRetryWithIngestion` will get longer and longer. Having a dedicated method such as `verifyIngestion` that calls those three different methods will make it look a lot cleaner in my opinion - also adding logs to indicate which of the checks fail will be beneficial for debugging